### ISSUE TITLE
Error building NGINX

### ISSUE THREAD

**Issue by adominik42** [laradock/laradock#397]
I get the following error when i try to build nginx container:
ERROR: Service 'caddy' failed to build: The command '/bin/sh -c apk update     && apk upgrade     && apk add tar curl' returned a non-zero code: 2.

**Comment by Mahmoudz** [comment]
Can you try now: pull the latest master branch and rebuild your container.

**Comment by adominik42** [comment]
After pulling latest master branch and rebuilding all my containers(just in case), at nginx step 6, same error again.

`Step 6 : RUN apk update     && apk upgrade     && apk add --no-cache bash     && adduser -D -H -u 1000 -s /bin/bash www-data     && rm /etc/nginx/conf.d/default.conf     && echo "upstream php-upstream { server ${PHP_UPSTREAM}:9000; }" > /etc/nginx/conf.d/upstream.conf
 ---> Running in 3a630407bd6d
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.4/main: temporary error (try again later)
WARNING: Ignoring APKINDEX.167438ca.tar.gz: No such file or directory
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.4/community: temporary error (try again later)
WARNING: Ignoring APKINDEX.a2e6dac0.tar.gz: No such file or directory
2 errors; 24 distinct packages available
ERROR: Service 'nginx' failed to build: The command '/bin/sh -c apk update     && apk upgrade     && apk add --no-cache bash     && adduser -D -H -u 1000 -s /bin/bash www-data     && rm /etc/nginx/conf.d/default.conf     && echo "upstream php-upstream { server ${PHP_UPSTREAM}:9000; }" > /etc/nginx/conf.d/upstream.conf' returned a non-zero code: 2
`

**Comment by appleboy** [comment]
I can't reproduce this issue.

```
$ docker-compose build --no-cache nginx
Building nginx
Step 1 : FROM nginx:alpine
 ---> 935bd7bf8ea6
Step 2 : MAINTAINER Mahmoud Zalt <mahmoud@zalt.me>
 ---> Running in d7750b987830
 ---> 63f2e2351aa1
Removing intermediate container d7750b987830
Step 3 : ADD nginx.conf /etc/nginx/
 ---> e1d10c567cba
Removing intermediate container 0a47d0c41a7f
Step 4 : COPY sites/*.conf /etc/nginx/sites-available/
 ---> c4b332496ad2
Removing intermediate container 48fa5483ad81
Step 5 : ARG PHP_UPSTREAM=php-fpm
 ---> Running in fa6f1161832d
 ---> 16d502c9142f
Removing intermediate container fa6f1161832d
Step 6 : RUN apk update     && apk upgrade     && apk add --no-cache bash     && adduser -D -H -u 1000 -s /bin/bash www-data     && rm /etc/nginx/conf.d/default.conf     && echo "upstream php-upstream { server ${PHP_UPSTREAM}:9000; }" > /etc/nginx/conf.d/upstream.conf
 ---> Running in 44215163b1ed
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
v3.4.5-5-g512f81f [http://dl-cdn.alpinelinux.org/alpine/v3.4/main]
v3.4.4-21-g75fc217 [http://dl-cdn.alpinelinux.org/alpine/v3.4/community]
OK: 5981 distinct packages available
(1/8) Upgrading musl (1.1.14-r11 -> 1.1.14-r14)
(2/8) Upgrading gd (2.2.1-r0 -> 2.2.3-r1)
(3/8) Upgrading libcrypto1.0 (1.0.2h-r1 -> 1.0.2j-r0)
(4/8) Upgrading libssl1.0 (1.0.2h-r1 -> 1.0.2j-r0)
(5/8) Upgrading libgpg-error (1.22-r0 -> 1.23-r0)
(6/8) Upgrading libgcrypt (1.7.0-r0 -> 1.7.0-r1)
(7/8) Upgrading busybox (1.24.2-r9 -> 1.24.2-r12)
Executing busybox-1.24.2-r12.post-upgrade
(8/8) Upgrading musl-utils (1.1.14-r11 -> 1.1.14-r14)
Executing busybox-1.24.2-r12.trigger
OK: 54 MiB in 24 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
(1/5) Installing ncurses-terminfo-base (6.0-r7)
(2/5) Installing ncurses-terminfo (6.0-r7)
(3/5) Installing ncurses-libs (6.0-r7)
(4/5) Installing readline (6.3.008-r4)
(5/5) Installing bash (4.3.42-r3)
Executing bash-4.3.42-r3.post-install
Executing busybox-1.24.2-r12.trigger
OK: 62 MiB in 29 packages
 ---> 9cb59554236c
Removing intermediate container 44215163b1ed
Step 7 : CMD nginx
 ---> Running in 5f12fc4a6555
 ---> 6b0fbc60d030
Removing intermediate container 5f12fc4a6555
Step 8 : EXPOSE 80 443
 ---> Running in cc84049e7991
 ---> 1d9523a11b18
Removing intermediate container cc84049e7991
Successfully built 1d9523a11b18
```

I think you can try to download `http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz` first.

**Comment by dmassiani** [comment]
I have the same error.

**Comment by appleboy** [comment]
@dmassiani What is your laradock version?

**Comment by dmassiani** [comment]
Last

**Comment by eddigei** [comment]
same problem

**Comment by helbing** [comment]
same problem too，but if enter the command `docker-compose up -d mysql nginx` again , it success

**Comment by mouyong** [comment]
i have the same problem, but i dont know how to deal with it, could you give me some ideas? thanks.

**Comment by kosilence** [comment]
same problem
`Step 5 : RUN apk update     && apk upgrade     && apk add --no-cache bash     && adduser -D -H -u 1000 -s /bin/bash www-data     && rm /etc/nginx/conf.d/default.conf     && echo "upstream php-upstream { server ${PHP_UPSTREAM}:9000; }" > /etc/nginx/conf.d/upstream.conf
 ---> Running in 24893df580bd
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
v3.4.6-66-g8f0ae71 [http://dl-cdn.alpinelinux.org/alpine/v3.4/main]
v3.4.6-33-g38ef2d2 [http://dl-cdn.alpinelinux.org/alpine/v3.4/community]
OK: 5976 distinct packages available
(1/3) Upgrading libcrypto1.0 (1.0.2j-r0 -> 1.0.2k-r0)
(2/3) Upgrading libssl1.0 (1.0.2j-r0 -> 1.0.2k-r0)
(3/3) Upgrading busybox (1.24.2-r12 -> 1.24.2-r13)
Executing busybox-1.24.2-r13.post-upgrade
Executing busybox-1.24.2-r13.trigger
OK: 54 MiB in 24 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
(1/5) Installing ncurses-terminfo-base (6.0-r7)
(2/5) Installing ncurses-terminfo (6.0-r7)
ERROR: Service 'nginx' failed to build: The command '/bin/sh -c apk update     && apk upgrade     && apk add --no-cache bash     && adduser -D -H -u 1000 -s /bin/bash www-data     && rm /etc/nginx/conf.d/default.conf     && echo "upstream php-upstream { server ${PHP_UPSTREAM}:9000; }" > /etc/nginx/conf.d/upstream.conf' returned a non-zero code: 137`

**Comment by mouyong** [comment]
# You can fix him like this：

```
FROM nginx:alpine

MAINTAINER Mahmoud Zalt <mahmoud@zalt.me>

ADD nginx.conf /etc/nginx/

ARG PHP_UPSTREAM=php-fpm

RUN sed -i 's/dl-cdn.alpinelinux.org/mirrors.aliyun.com/' /etc/apk/repositories 
RUN    apk update \
    && apk upgrade \
    && apk add --no-cache bash \
    && adduser -D -H -u 1000 -s /bin/bash www-data \
    && rm /etc/nginx/conf.d/default.conf \
    && echo "upstream php-upstream { server ${PHP_UPSTREAM}:9000; }" > /etc/nginx/conf.d/upstream.conf


CMD ["nginx"]

EXPOSE 80 443

```

**Comment by philtrep** [comment]
@mouyong Awesome! Could you submit a PR? I'd prefer it if you got credit for the fix 😄

**Comment by mouyong** [comment]
@philtrep i'm clone it in my space, and submit a PR.

**Comment by mouyong** [comment]
@philtrep okay, I'm clone it in my space, and submit a PR.

**Comment by philtrep** [comment]
Closed by #609. Thanks @mouyong !

**Comment by NisalG** [comment]
I was able to fix this based on comment by @kosilence
I changed laradock/.env as below:
### NGINX Container
#NGINX_HOST_HTTP_PORT=80
#NGINX_HOST_HTTPS_PORT=443
NGINX_HOST_HTTP_PORT=8083
NGINX_HOST_HTTPS_PORT=4431
